src/common/sha2/sha2.c

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: Public domain
+#include <stdio.h>
+
+#include <oqs/common.h>
+
+#include "sha2.h"
+#include "sha2_local.h"
+
+#if defined(OQS_DIST_ARM64v8_BUILD)
+#define C_OR_NI(stmt_c, stmt_ni) { \
+    do { \
+        if (OQS_CPU_has_extension(OQS_CPU_EXT_ARM_SHA2)) {  \
+            stmt_ni; \
+        } else { \
+            stmt_c; \
+        } \
+    } while(0)
+#elif defined(OQS_USE_ARM_SHA2_INSTRUCTIONS)
+#define C_OR_NI(stmt_c, stmt_ni) \
+    stmt_ni
+#else
+#define C_OR_NI(stmt_c, stmt_ni) \
+    stmt_c
+#endif
+void OQS_SHA2_sha224_inc_init(OQS_SHA2_sha224_ctx *state) {
+	oqs_sha2_sha224_inc_init_c((sha224ctx *) state);
```

Comment:
Why are these casts necessary?

Reply:
I added them because the generic OQS_SHA2_sha*_ctx structs contain a void pointer, and I was getting typing errors for using the more specific sha*ctx structs which contain a uint8_t pointer. This seemed like the easiest way to keep the OQS_SHA2_sha*_ctx structs general while not having to cast the void pointer to a uint_8t every access. Do you have any recommendations on how to do this in a simpler/cleaner way?

Reply:
No I guess that makes sense.